receiver/podmanreceiver/receiver.go

```diff
@@ -87,3 +170,56 @@ func (r *receiver) scrape(context.Context) (pdata.Metrics, error) {
 	}
 	return md, nil
 }
+
+func (r *receiver) handleEvents(ctx context.Context, eventBackoff *backoff.ExponentialBackOff) error {
+	c, err := r.clientFactory(r.set.Logger, r.config)
+	if err != nil {
+		r.set.Logger.Error("error fetching/processing events", zap.Error(err))
+		return err
+	}
+	r.client = c
+
+	// Fetch the response from the endpoint
+	response, err := r.client.getEventsResponse(ctx)
+	if err != nil {
+		r.set.Logger.Error("error while fetching events", zap.Error(err))
+		return err
+	}
+
+	dec := json.NewDecoder(response.Body)
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			// Translate the response to the events format.
+			decodedEvent, err := decodeEvents(dec)
+			if err != nil {
+				// context will be cancelled only in the case when shutdown will be called. So returning gracefully.
+				if errors.Is(err, context.Canceled) {
+					return nil
+				}
+				r.set.Logger.Error("error decoding the event", zap.Error(err))
+				if err = response.Body.Close(); err != nil {
+					r.set.Logger.Error("error while closing the connection", zap.Error(err))
+					return err
+				}
+				return err
+			}
+
+			// Translate the events into the pdata.logs format
+			ld, err := translateEventsToLogs(r.set.Logger, decodedEvent)
+			if err != nil {
+				r.set.Logger.Error("error translating event to log", zap.Error(err))
+				return err
+			}
+
+			err = r.logsConsumer.ConsumeLogs(ctx, ld)
+			if err != nil {
+				r.set.Logger.Error("error transferring it to the next component", zap.Error(err))
+				return err
+			}
+			eventBackoff.Reset()
+		}
+	}
```

Comment:
Either I don't understand this logic or it's wrong. Are you fetching the events payload only once and then sending the same logs from that payload in the infinite loop?

Reply:
We are making an API call only once. But in that API call, we are giving the parameter `stream=true`. This parameter gives the capability to stream the event response.

The Buffer is provided in the response of the API call. We are reading that buffer in the for loop to get the new events. Buffer will be updated as soon as any events will be generated in the podman.

We have two ways to get events. Both are using `/events` API call:
- If we provide `stream=false`, then the response will be of all the events till the time of API call and the response will be closed. It will not be streaming events. 
- If we provide `stream=true`, then the response will be a buffer, which will be updated as and when the event occurs in the podman.

Documentation(link)

Reply:
Thanks for clarifying! It makes sense now. `scraperhelper` is not needed then

Did you build and test this?

Reply:
Yes, we have tested.